components/Image/image-responsive.tsx

```diff
@@ -0,0 +1,288 @@
+    let canMouseX1: number
+    let canMouseY1: number
+    let canMouseX2: number
+    let canMouseY2: number
+    if (e?.nativeEvent?.clientX && e?.nativeEvent?.clientY) {
+      //canMouseX = parseInt(e.clientX - cfg.offsetX)
+      canMouseX = e.nativeEvent.clientX - offsetLeft
+      canMouseY = e.nativeEvent.clientY - offsetTop
+    } else if (e?.nativeEvent?.targetTouches) {
+      canMouseX1 = e.nativeEvent.targetTouches.item(0)?.clientX - offsetLeft
+      canMouseY1 = e.nativeEvent.targetTouches.item(0)?.clientY - offsetTop
+      if (e?.nativeEvent?.touches?.length > 1) {
+        //To get the center of the 'pinch' we to 'average' the two points
+        canMouseX2 = e.nativeEvent.targetTouches.item(1)?.clientX - offsetLeft
+        canMouseY2 = e.nativeEvent.targetTouches.item(1)?.clientY - offsetTop
+        canMouseX = Math.round((canMouseX1 + canMouseX2) / 2)
+        canMouseX = Math.round((canMouseY1 + canMouseY2) / 2)
+      } else {
+        canMouseX = canMouseX1
+        canMouseY = canMouseY1
+      }
+    }
+    return {
+      canMouseX,
+      canMouseY,
+    }
+  }
+
+  /* MOUSE TOUCH HANDLERS
+  variables canMouseX/Y refer to the CURRENT coordinates of the mouse or touch for current detected event on the 'canvas'.
+  canMouseX/Y coordinates are compared to the last stored coordinates (oldXCoord,oldYCoord) for purposed of
+  determining the direction of a drag. In the case of pinch, two sets of current coordinates measure a distance between
+  those two points and that to a previously stored distance to determine a pinch "in" or "out"
+  */
+
+  //When Mouse button up (touch released) reset to orginal setting
+  const handleMouseUp = (e: any) => {
+    setIsScaling(false)
+    setIsDragging(false)
+    setPanRate(1)
+    setTouchDist(0)
+  }
+
+  const handleMouseLeave = (e: any) => {
+    //for now, leaving the image does the same as release mouse or touch
+    handleMouseUp(e)
+  }
+
+  const handleMouseDown = (e: any) => {
+    const { canMouseX, canMouseY } = setCoordinates(e)
+    canMouseX ? setOldXCoord(canMouseX) : setOldXCoord(0)
+    canMouseY ? setOldYCoord(canMouseY) : setOldYCoord(0)
+    e.preventDefault()
+    setIsDragging(true)
+    if (e?.targetTouches) {
+      //test for more than one touch to detect pinch
+      if (e?.nativeEvent?.touches?.length > 1) {
+        setTouchDist(distance(e)) //set pinch distance to test for zooming in/out later
+        setIsScaling(true) //detected pinch so we are now scaling...
+        setIsDragging(false) //...and we are not dragging
+      } else {
+        // otherwise, a single detected 'down' touch means we are dragging/panning not scaling
+        setIsScaling(false)
+        setIsDragging(true)
+      }
+    }
+  }
+
+  // Double click acts as quick scale option, gives PC users scaling ability, and allows for quick resetting back
+  // to scale = 1 (scaling cycle back in this handler to 1 after max reached)
+  const handleDoubleClick = (e: any) => {
```

Comment:
Let's nail down the type here. Should be a `React.MouseEvent`

Reply:
Same as above